code/modules/mob/living/simple_animal/abnormality/he/singing_machine.dm

```diff
@@ -130,49 +161,96 @@ Finally, an abnormality that DOESN'T have to do any fancy movement shit. It's a
+		to_chat(addict, span_warning("You can hear it again... it needs more..."))
+		if(!addict.sanity_lost)
+			continue
+		applySpecialInsanity(addict)
+
+/mob/living/simple_animal/hostile/abnormality/singing_machine/proc/addAddict(mob/living/carbon/human/target)
+	target.apply_status_effect(STATUS_EFFECT_MUSIC)
+	to_chat(target, span_nicegreen("There's something about that sound..."))
+	musicalAddicts |= target
+	SEND_SOUND(target, 'sound/abnormalities/singingmachine/addiction.ogg')
+	addtimer(CALLBACK(src, .proc/removeAddict, target), 5 MINUTES)
+
 /mob/living/simple_animal/hostile/abnormality/singing_machine/proc/removeAddict(mob/living/carbon/human/addict)
 	if(addict)
 		musicalAddicts -= addict // Your five minutes are over, you're free.
 
+/mob/living/simple_animal/hostile/abnormality/singing_machine/proc/startPlaying()
+	particle_note = new(get_turf(src))
+	particle_blood = new(get_turf(src))
+	grindNoise = new(list(src), TRUE)
+	musicNoise = new(list(src), TRUE)
+	driveInsane(musicalAddicts)
+	playStatus = PLAYING
+	addtimer(CALLBACK(src, .proc/stopPlaying), playLength) // This is the callback from earlier.
+
 /mob/living/simple_animal/hostile/abnormality/singing_machine/proc/stopPlaying()
+	playStatus = SILENT // This exists solely because I needed to call it via a callback.
 	if(grindNoise)
 		QDEL_NULL(grindNoise)
 	if(musicNoise)
 		QDEL_NULL(musicNoise)
-	playStatus = 0 // This exists solely because I needed to call it via a callback.
+	if(particle_grind)
+		particle_grind.fadeout()
+	if(particle_note_broken)
+		particle_note_broken.fadeout()
+	if(particle_note)
+		particle_note.fadeout()
+	if(particle_blood)
+		particle_blood.fadeout()
 
 /mob/living/simple_animal/hostile/abnormality/singing_machine/proc/eatBody(mob/living/carbon/human/user)
 	user.gib()
 	stopPlaying()
 	playsound(src, 'sound/abnormalities/singingmachine/swallow.ogg', 80, 0, 3)
-	cleanliness = "bloody"
-	icon_state = "singingmachine_closed_[cleanliness]"
+	machine_hinge.bloodySelf()
+	icon_state = "singingmachine_bloody"
 	icon_living = icon_state
-	update_icon()
-	driveInsane(musicalAddicts)
-	playStatus = 2
+	machine_hinge.moveHinge(0 ,3 SECONDS, 0, 0)
+	addtimer(CALLBACK(src, .proc/startPlaying), 3 SECONDS)
 	datum_reference.qliphoth_change(2)
-	grindNoise = new(list(src), TRUE)
-	musicNoise = new(list(src), TRUE)
-	addtimer(CALLBACK(src, .proc/stopPlaying), playLength) // This is the callback from earlier.
+
+/mob/living/simple_animal/hostile/abnormality/singing_machine/proc/applySpecialInsanity(mob/living/carbon/human/addict)
+	QDEL_NULL(addict.ai_controller)
+	addict.ai_controller = /datum/ai_controller/insane/murder/singing_machine
+	addict.InitializeAIController()
+	addict.add_overlay(mutable_appearance('ModularTegustation/Teguicons/tegu_effects.dmi', "singing_smile", -MUTATIONS_LAYER))
 
 /mob/living/simple_animal/hostile/abnormality/singing_machine/proc/driveInsane(list/addicts)
```

Comment:
This proc is specifically handed a list accessible by THIS abnormality. There is no reason for it to take a list as input and not just instead reference the list it gets handed.

Reply:
Oops, will solve